operator/src/main/java/oracle/kubernetes/operator/steps/ReadHealthStep.java

```diff
@@ -89,13 +103,110 @@ public NextAction apply(Packet packet) {
     V1Pod pod = info.getServerPod(serverName);
     if (service != null) {
       Step getClient =
-          HttpClient.createAuthenticatedClientForServer(
-              namespace, secretName, new ReadHealthWithHttpClientStep(service, pod, getNext()));
+          getEncodedCredentialsForServer(
```

Comment:
The EncodeCredentialsForServerStep is just creating a readSecret step and then call doNext(). Can it be removed and instead we call SecretHelper.getSecretData() here?

Reply:
Fixed. Thanks.